ui/components/component-library/button/button.types.ts

```diff
@@ -23,15 +23,15 @@ type ButtonPropsByVariant = {
   [ButtonVariant.Primary]: {
     variant?: ButtonVariant.Primary;
     size?: ValidButtonSize; // Allows for only ButtonSize.Sm, ButtonSize.Md, ButtonSize.Lg
-  } & Omit<ButtonPrimaryStyleUtilityProps, 'size'>;
+  } & Omit<ButtonPrimaryStyleUtilityProps, 'size' | 'variant'>;
```

Comment:
After updating ButtonBase to extend TextStyleUtilityProps there was a clash of `variant` I think @garrettbear brought this up before. The workaround here is if the `Text` `variant` needs to be changed for a button which in most cases it shouldn't we can suggest the use of the variant components `ButtonPrimary`, `ButtonSecondary`, `ButtonLink` instead